### ISSUE TITLE
Useless function call is not removed by plugin-remove-useless-functions

### ISSUE THREAD

**Issue by mrkosima** [coderaiser/putout#113]
Hi there 👋  
It could be that the plugin I'm looking for exists, but I can't find the one.
I have the next statement:

```js
const test = (fn, value) => {
  return (a => fn(a))(value);
}

const test(console.log, 42);
```

I want it to be transformed to 

```js
const test = (fn, value) => {
  return fn(value);
}

const test(console.log, 42);
```

I expected that [plugin-remove-useless-functions](https://github.com/coderaiser/putout/tree/master/packages/plugin-remove-useless-functions) does the thing, but it doesn't. 

What's the best way I can solve the issue?

**Comment by coderaiser** [comment]
How many such code do you have in your codebase?

**Comment by mrkosima** [comment]
Yes, this pseudo code is a result of another automated cleanup. 
To give you more understanding, it's about express middleware:

```js
router.use(
    ... // some middlewares
    (err, req, res, next) => {
        const { customValue } = res.locals;
        return myMiddleware({
           customParam: customValue, 
           otherParams: 42,
        })(err, req, res, next),
    }
)
```

Later on the `customValue` became no longer needed, and the code was transformed (with codemods) into:

```js
router.use(
    .... // some middlewares
    (err, req, res, next) => myMiddleware({ otherParams: 42 })(err, req, res, next),
)
```

But I want it to become:

```js
router.use(
    .... // some middlewares
    myMiddleware({ otherParams: 42 }),
)
```

And there a many cases like this (> 100).

Based on your suggestion, I just checked this "replace" - and it does what I need:
```js
export const replace = () => ({
    '(__a, __b, __c, __d) => __m(__x)(__a, __b, __c, __d)': '__m(__x)',
});
```

I think, this issue can be closed, unless you're not planning to support the automated fix with the default plugin.


Thank you a lot for the help!

**Comment by coderaiser** [comment]
I remembered, here is [the answer](https://github.com/coderaiser/putout/tree/master/packages/plugin-remove-useless-functions#open-questions) to your question :).
Some code can break after such transformation, so it's not too common to add it to a plugin.

Great that everything is worked out :)!

**Comment by coderaiser** [comment]
Just landed to [@putout/plugin-remove-iife](https://github.com/coderaiser/putout/tree/v27.8.0/packages/plugin-remove-iife#readme), it's more related :).

You can try it in 🐊**Putout** [`v27.8.0`](https://github.com/coderaiser/putout/releases/tag/v27.8.0) 🎉

**Comment by mrkosima** [comment]
Just checked how the `remove-iife` plugin works. 
Here some extra cases I checked:

1. Unused params
```js
((a) => fn(42))(value);
```
**Result**: `fn(value)`
**Expected**: `fn(42)` or `noTransform()`.

2. The previous example can be extended with:
```js
((a) => fn(a, 42))(value);
```
**Expected**: `fn(value, 42)` or `noTransform()`.

3. Used variables within a function body
```js
((a) => fn({customProp: a.x}))(value);
```
**Result**: `fn(value)`
**Expected**: `fn({customProp: value.x})` or `noTransform()`

What do you think?

**Comment by coderaiser** [comment]
Thanks for exhaustive examples :)! Just improved support of linked arguments [`__args__a`](ttps://github.com/coderaiser/putout/tree/master/packages/compare#__args-__args_a) in `@putout/compare` [`v9.4.0`](https://www.npmjs.com/package/@putout/compare) 🎉!
Also updated examples of 🦎[**PutoutScript**](https://github.com/coderaiser/putout/blob/master/docs/putout-script.md#__args-template-value).

Please reinstall 🐊**Putout**. Is it works for you?
